core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -3533,6 +3533,30 @@ protected RelDataType validateSelectList(
         boolean matches = aggScope.checkAggregateExpr(selectItem, true);
         Util.discard(matches);
       }
+    } else {
+      // For non-aggregating select check nested window aggregates reference only
+      // GROUP BY columns as partition columns
+      for (SqlNode selectItem : expandedSelectItems) {
+        // Find the OVER node in the window aggregate
+        SqlNode windowAgg = overFinder.findAgg(selectItem);
+        if (windowAgg != null) {
+          // Find the nested aggregate, if any
+          AggFinder nestedAggFinder = new AggFinder(opTab, false, false, aggFinder);
+          SqlNode nestedAgg = nestedAggFinder.findAgg(((SqlCall) windowAgg).operand(0));
+          SqlNode window = ((SqlCall) windowAgg).operand(1);
+          // If nested agg present, the partition list should be empty since no
+          // GROUP BY present
+          if (nestedAgg != null
+                  && window instanceof SqlWindow) {
+            SqlWindow sqlWindow = (SqlWindow) window;
+            SqlNodeList partitionList = sqlWindow.getPartitionList();
```

Comment:
In addition to the partition-by list, the order-by list should also be examined for same conditions.  (i.e for queries such as  'select avg(sum(sal)) over (partition by deptno order by empno) from emp'